pasqal_cloud/utils/models.py

```diff
@@ -0,0 +1,150 @@
+            cls.convert_str_to_uuid(item)
+        return values
+
+
+class JobFilters(BaseFilters):
+    """
+    Class to provide filters for querying jobs.
+
+    Setting a value for any attribute of this class will add a filter to the query.
+
+    When using several filters at the same time, the API will return elements who pass
+    all filters at the same time:
+        - If the filter value is a single element, the API will return jobs whose
+          attribute matches the provided value.
+        - If the filter value is a list, the API will return jobs whose value for
+          this attribute is contained in that list.
+
+    Attributes:
+        id: Filter by job IDs.
+        project_id: Filter by project IDs.
+        user_id: Filter by user IDs.
+        batch_id: Filter by batch IDs.
+        status: Filter by job statuses.
+        min_runs: Minimum number of runs.
+        max_runs: Maximum number of runs.
+        errors: Filter by presence of errors.
+        start_date: Retry jobs created at or after this datetime.
+        end_date: Retry jobs created at or before this datetime.
+    """
+
+    id: Optional[Union[List[Union[UUID, str]], Union[UUID, str]]] = None
+    project_id: Optional[Union[List[Union[UUID, str]], Union[UUID, str]]] = None
+    user_id: Optional[Union[List[str], str]] = None
+    batch_id: Optional[Union[List[Union[UUID, str]], Union[UUID, str]]] = None
+    status: Optional[Union[List[JobStatus], JobStatus]] = None
+    min_runs: Optional[int] = None
+    max_runs: Optional[int] = None
+    errors: Optional[bool] = None
+    start_date: Optional[datetime] = None
+    end_date: Optional[datetime] = None
+
+    @field_validator("id", "project_id", "user_id", "batch_id", "status", mode="before")
+    @classmethod
+    def single_item_to_list_validator(cls, values: Dict[str, Any]) -> Any:
+        return cls.convert_to_list(values)
+
+    @field_validator("id", "project_id", "batch_id", mode="after")
+    @classmethod
+    def str_to_uuid_validator(cls, values: Dict[str, Any]) -> Dict[str, UUID]:
+        for item in values:
+            cls.convert_str_to_uuid(item)
+        return values
+
+
+class PaginationParams(BaseFilters):
+    """
+    Class providing parameters for paginating data query results.
+
+    Attributes:
+        offset: The starting index of the data to return (must be 0 or greater).
+        limit: The maximum number of data to return (must be greater than 0 and
+               less than or equal to 100).
+    """
+
+    offset: int = Field(default=0, strict=True, ge=0)
+    limit: int = Field(default=100, strict=True, gt=0, le=100)
+
+
+class PaginatedResponse(BaseModel):
+    """
+    Class representing a paginated response structure.
+
+    Attributes:
+        total: The total number of data matching the query.
+        results: A list of data that match the query and pagination parameters
+                 provided.
+    """
+
+    total: int
+    results: List[Any]
```

Comment:
I think it's useful to expose the offset to the user so it helps them iterate over pages easily
```suggestion
    offset: int
    results: List[Any]
```